### DISCUSSION TITLE
Fit summary (AIC) information AutoARIMA

### DISCUSSION THREAD

**Discussion by Gesh96** [Nixtla/statsforecast#273] [Q&A] [answered]
Hello!

Could you please tell me whether a feature like summary() in pmdarima is being developed? The main thing that I would be interested in having is the option to extract the AIC of the fitted AutoARIMA model, so that I could test various models in-sample and choose the one with the lowest AIC.

Looking forward to your answer, thanks a lot in advance!

P.S. The AutoARIMA seems to forecast extremely accurately, great job!

**Comment by kdgutier** [comment, marked as answer]
Hi @Gesh96,

We are working on the summary method, meanwhile the `AutoARIMA`, `AutoCES` and `ETS` models already perform model selection based on the AIC/BIC criterion within their family of models. A momentary solution if you want to compare AIC across methods is to use the insample predictions to generate the metrics: 

```
from statsforecast.core import StatsForecast
from statsforecast.models import AutoARIMA, ETS

fcst = StatsForecast(df=Y_train_df,
                     models=[AutoARIMA(seasonal_length='your_seasonality')],
                     freq='your_freq', n_jobs=-1)
                     
fcsts_df  =  fcst.forecast(h=h, fitted=True)
insample_fcsts_df = fcst.forecast_fitted_values() # y and y_hat for insample metrics
```

If you are considering selecting the best model between `AutoARIMA`, `AutoCES` and `ETS`, ensembling them with median or mean predictions works really well.

**Reply by Gesh96** [reply]
Thank you very much for the answer!
